src/objects.rs

```diff
@@ -786,7 +786,10 @@ impl FromStr for Time {
         };
         let hours: u32 = hours.parse()?;
         let minutes: u32 = minutes.parse()?;
-        let seconds: u32 = seconds.parse()?;
+        let seconds: u32 = match seconds.split('.').next() {
+            Some(seconds) => seconds.parse()?,
+            _ => return Err(TimeError::WrongValue),
+        };
```

Comment:
This is kind of misleading since the `_` will never be reached as far as I understand (`split()` always return at least one element). An alternative solution would be to parse a sub-slice of `seconds` from start until the character dot `.` or until the end.

```suggestion
        let seconds: u32 = seconds[..seconds.find('.').unwrap_or(seconds.len())].parse()?;
```

or maybe more readable in two lines

```suggestion
        let last_second_number = seconds.find('.').unwrap_or(seconds.len());
        let seconds: u32 = seconds[..last_second_number].parse()?;
```

Reply:
Note that this is a `.unwrap_or()` (which will not `panic!()` in case of `None`), not an `.unwrap()` (which will `panic!()` in case of `None`).

Reply:
I know but I didn’t want to use unwrap()

Reply:
Note that this is a `.unwrap_or()` (which will not `panic!()` in case of `None`), not an `.unwrap()` (which will `panic!()` in case of `None`).

Reply:
Another option if you prefer the `match` would be to do the following.

```suggestion
        let seconds: u32 = match seconds.split('.').next() {
            Some(seconds) => seconds.parse()?,
            _ => unreachable!(".split() always return at least one element"),
        };
```

Reply:
I know but in my use I didn't want any default value

Reply:
done in 2720d2b7feba5aa9639dc27a2dc37e6e46ac0ee7